app/services/katello/pulp3/repository.rb

```diff
@@ -80,10 +80,25 @@ def update_distribution(path)
 
       def refresh_distributions
         paths.map do |prefix, path|
-          if distribution_reference(path)
-            update_distribution(path)
+          dist_ref = distribution_reference(path)
+          if dist_ref
+            if prefix == :http
+              if repo.root.unprotected
+                update_distribution(path)
+              else
+                result = delete_distribution(dist_ref.href)
+                dist_ref.destroy
+                result
+              end
+            else
+              update_distribution(path)
+            end
           else
-            create_distribution(prefix, path)
+            if prefix == :http
+              create_distribution(prefix, path) if repo.root.unprotected
+            else
+              create_distribution(prefix, path)
+            end
           end
```

Comment:
Copied from @jjeffers update PR since there were issues with the distribution creation.